addons/account_analytic_purchase/models/purchase_order_line.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    def _prepare_compute_all_values(self):
+        res = super(PurchaseOrderLine, self)._prepare_compute_all_values()
+        default_analytic_account = self.env['account.analytic.default'].account_get(self.product_id.id, self.partner_id.id, self.order_id.user_id.id, fields.Date.today())
+        if default_analytic_account:
+            res.update({'account_analytic_id': default_analytic_account.analytic_id.id})
+            res.update({'analytic_tag_ids': [(6, 0, default_analytic_account.analytic_tag_ids.ids)]})
```

Comment:
Because call to update one more again?
Into single update can you pass account_analytic and analytic_tag_ids

Reply:
If you are going to comment about small improvements like that at least do it right (the results are very similar without the dict initialization):
Here is what I do
```
$ python -m timeit 'd={"k":1, "l":1}; d.update({"k":2}); d.update({"l":2})'
1000000 loops, best of 3: 0.433 usec per loop
```

Here is what you propose
```
$ python -m timeit 'd={"k":1, "l":1}; d.update({"k":2, "l":2})'
1000000 loops, best of 3: 0.276 usec per loop
```

But here is the real solution
```
$ python -m timeit 'd={"k":1, "l":1}; d["k"] = 2; d["l"] = 2'
10000000 loops, best of 3: 0.115 usec per loop
```

This is a very small optimization though, the real bottleneck here is the creation of the record in the database afterwards, not the manipulations on a python dict.

I will do the changes :+1: